fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java

```diff
@@ -226,6 +265,52 @@ public void alterCatalog(AlterCatalogStmt stmt) {
         }
     }
 
+    private void alterCatalog(Catalog catalog, Map<String, String> properties, boolean isReplay) throws DdlException {
+        Map<String, String> alterProperties = new HashMap<>(properties.size());
+        Map<String, String> oldProperties = catalog.getConfig();
+
+        for (String confName : properties.keySet()) {
+            String oldVal = oldProperties.get(confName);
+            String newVal = properties.get(confName);
+            if (!oldProperties.containsKey(confName) || !Objects.equals(oldVal, newVal)) {
+                alterProperties.put(confName, newVal);
+            }
+        }
+
+        if (alterProperties.isEmpty()) {
+            return;
+        }
+
+        if (needRecreateCatalog(catalog, alterProperties)) {
+            // recreate catalog
+            reCreatCatalog(catalog, alterProperties, isReplay);
+        } else {
+            String catalogName = catalog.getName();
```

Comment:
Why do we need to handle the logic for Ranger separately? Can't we reuse the logic reCreatCatalog?

Reply:
Rebuilding a catalog is a heavyweight operation. Therefore, we handle it in two scenarios: we avoid rebuilding when unnecessary to prevent any impact on queries.

Reply:
The operation of  alter catalog  is rarely performed. I don't think it's worth increasing the complexity of the code logic for this.

Reply:
ok